docs/website/versioned_docs/version-3.0.0/command-reference/add-binding.md

```diff
@@ -2,41 +2,50 @@
 
 Currently, it only allows connecting to the operator-backed services which support binding via the Service Binding Operator.
 To know about the operators supported by the Service Binding Operator, read their [README](https://github.com/redhat-developer/service-binding-operator#known-bindable-operators).
 
-## Pre-requisites
+## Running the Command
+
+### Pre-requisites
 * A directory containing a Devfile; if you don't have one, see [odo init](init.md) on obtaining a devfile.
 * A cluster with the Service Binding operator installed, along with the operator whose service you need to bind to
 
-## Interactive Mode
+### Interactive Mode
 In the interactive mode, you will be guided to choose:
 * a service from the list of bindable service instances as supported by the Service Binding Operator,
-* option to bind the service as a file,
+* option to bind the service as a file; see [Understanding Bind as Files](#understanding-bind-as-files) for more information on this.
 * a name for the binding.
 
 ```shell
-# Add binding between service named 'myservice',
-# and the component present in the working directory in the interactive mode
+# Add binding between a service, and the component present in the working directory in the interactive mode
 odo add binding
 ```
 
-## Non-interactive mode
+### Non-interactive mode
 In the non-interactive mode, you will have to specify the following required information through the command-line:
 * `--service` flag to specify the service you want to bind to,
-* `--name` flag to specify a name for the binding,
+* `--name` flag to specify a name for the binding; see [Understanding Bind as Files](#understanding-bind-as-files) for more information on this.
 * `--bind-as-files` flag to specify if the service should be bound as a file; this flag is set to true by default.
 
 
 ```shell
-# Add binding between a service named 'myservice',
+# Add binding between a service named 'cluster-sample',
 # and the component present in the working directory in the non-interactive mode
 odo add binding --name mybinding --service myRedisService.Redis
 ```
 
-### Formats supported by the `--service` flag
+#### Understanding Bind as Files
+To connect your component with a service, you need to store some data(e.g. username, password, host address) on your component's container.
+If the service is bound as files, this data will be written to a file and stored on the container, else it will be exported to Environment Variables inside the container.
+
+Note that every piece of data is stored in its own individual file or environment variable.
+For example, if your data includes a username, and password, then 2 separate files, or 2 environment variables will be created to store them both.
```

Comment:
```suggestion
#### Understanding Bind as Files
To connect your component with a service, you need to store some data (e.g. username, password, host address) on your component's container.
If the service is bound as files, this data will be written to a file and stored on the container, else it will be injected as Environment Variables inside the container.

Note that every piece of data is stored in its own individual file or environment variable.
For example, if your data includes a username and password, then 2 separate files, or 2 environment variables will be created to store them both.
```